### PR TITLE
Game logic

### PR DESCRIPTION
**Pull request by flov3rh**

### CODE REVIEW

File: lib/tic_tac_toe.rb [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+          (@board[1] == 'O' && @board[4] == 'O' && @board[7] == 'O') ||
+          (@board[2] == 'O' && @board[5] == 'O' && @board[8] == 'O') ||
+          (@board[0] == 'O' && @board[4] == 'O' && @board[8] == 'O') ||
+          (@board[2] == 'O' && @board[4] == 'O' && @board[6] == 'O')
+      @main.print_winner(@player_two)
+      true
+    elsif @moves == 9
+      @main.print_draw
+      true
+    end
```
Comment (stratospherique) [inline lib/tic_tac_toe.rb]: Your checkwin method could have been implemented in a more robust manner:
```ruby
def check_win(player_symbol)
    # for rows
    all_equal(board[0..2], player_symbol) ||
    all_equal(board[3..5], player_symbol) ||
    all_equal(board[6..8], player_symbol)

    # and analogical as above for columns and diagonals
  end
```

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Great work, your game is working fine :clap: . However, some requirements needs to be implemented in order for us to approve your project. check :arrow_down: 

##### Implementation details

* [x] The project uses at least two classes (Usually it uses three: Game, Board and Player). The code of the project should be logically divided between classes.
* [x] The code should respect the single responsibility rule for methods (i.e. each method does exactly one thing).
* [x] DRY rule should be respected. E.g. students tend to check winning position for X and O separately instead of creating a method and reuse it. check the comment code below :arrow_down: 

### That's it :relaxed: . When finished, kindly resubmit another code review request :pray: 

**Ahmed Mahfoudh**

File: bin/main.rb [hunk 2]
```diff
@@ -2,40 +2,54 @@
 
 # frozen_string_literal: true
 
+require './lib/tic_tac_toe.rb'
```
Comment (wathigo) [inline bin/main.rb:5]: **[OPTIONAL]**
* We can use `require_relative` instead of `require`. `require_relative` is the Ruby convention for importing files.

**Review by wathigo — APPROVED** [review]
### Hi @flov3rh  and @jcromerohdz  :wave: 
Great work implementing the requested changes.

**[OPTIONAL]** 
* We can implement play again functionality after a win or a draw.

**This is approved. You may move on :next_track_button:**  

_Good luck!
Reviewed by: **Simon Wathigo**_

**Pull request merged**